Assets/SequenceSDK/WaaS/Tests/EOAWalletToSequenceWalletTest.cs

```diff
@@ -0,0 +1,279 @@
+
+            if (txreturn is SuccessfulBatchTransactionReturn successfulBatchTransactionReturn)
+            {
+                foreach (var item in successfulBatchTransactionReturn.SuccessfulTransactionReturns)
+                {
+                    Assert.IsNotNull(item);
+                    Assert.IsNotNull(item.txHash);
+                    Assert.IsNotEmpty(item.txHash);
+                }
+            }
+            else if (txreturn is SuccessfulTransactionReturn successfulTransactionReturn)
+            {
+                Assert.IsNotEmpty(successfulTransactionReturn.txHash);
+            }
+            else if (txreturn is FailedTransactionReturn failedTransactionReturn)
+            {
+                Assert.Fail("Failed transaction return: "+ failedTransactionReturn.error );
+            }
+            else if (txreturn is FailedBatchTransactionReturn failedBatchTransactionReturn)
+            {
+                foreach (var item in failedBatchTransactionReturn.FailedTransactionReturns)
+                {
+                    Assert.Fail("Failed transaction return: " + item.error);
+                }
+                foreach (var item in failedBatchTransactionReturn.SuccessfulTransactionReturns)
+                {
+                    Assert.IsNotNull(item);
+                    Assert.IsNotNull(item.txHash);
+                    Assert.IsNotEmpty(item.txHash);
+                }
+            }
+        }
+
+        void OnTransactionSent(TransactionReturn txReturn)
+        {
+            Debug.Log(txReturn.ToString());
+        }
+        void OnContractDeployed(SuccessfulContractDeploymentReturn deploymentReturn)
+        {
+            Debug.Log(deploymentReturn.ToString());
+        }
+
+        async Task<ERC20> TestDeployERC20(string byteCode)
+        {
+            try
+            {
+                var deployedERC20return = await adapter.DeployContract(chain, byteCode, "0");
+
+                if (deployedERC20return is SuccessfulContractDeploymentReturn successfulContractDeploymentReturn)
+                {
+                    return new ERC20(successfulContractDeploymentReturn.DeployedContractAddress);
+                }
+                else if (deployedERC20return is FailedContractDeploymentReturn failedContractDeploymentReturn)
+                {
+                    throw new Exception("ContractDeployment failed: " + failedContractDeploymentReturn.Error);
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new Exception($"Error occurred during ERC20 contract deployment: {ex.Message}");
+            }
+            throw new Exception("Failed deployment transaction.");
+        }
+
+        async Task<SendERC20> MakeERC20Transaction(ERC20 erc20)
+        {
+            try
+            {
+                TransactionReceipt receipt = await erc20.Mint(adapter.GetWalletAddress(), BigInteger.Parse("1")).SendTransactionMethodAndWaitForReceipt(wallet, client);
```

Comment:
Here, you're using an EOA wallet to call the mint function. Instead, let's use the adapter and create a WaaS/SequenceWallet RawTransaction using the ERC20 object to encode the data bytes. This is a flow that isn't currently tested